products/pdt/pdtlisten/src/lib.rs

```diff
@@ -241,29 +245,28 @@ pub async fn listen_bq(bq_project_id: &str, bq_dataset_id: &str, api_url: &str)
 
     let provider = Provider::<Http>::try_from(api_url)?;
 
-    let mut stream = provider
-        .watch_blocks()
-        .await?
-        .map(|hash: H256| get_block_by_hash(hash, &provider))
-        .buffered(MAX_TASKS);
+    let stream = listen_blocks(&provider);
+    pin!(stream);
 
-    while let Some(block) = stream.next().await {
-        match block {
-            Ok(block) => {
-                // OK, time to deal with bigquery
-                if bq_importer.buffers.is_none() {
-                    bq_importer.reset_buffer(&zilliqa_bq_proj).await?;
-                }
-                // convert our blocks and insert it into our buffer
-                match {
-                    let (bq_block, bq_txns) = convert_block_and_txns(&block)?;
-                    bq_importer.insert_into_buffer(bq_block, bq_txns)
-                } {
-                    Ok(_) => (),
-                    Err(err) => {
-                        eprintln!("conversion to bq failed due to {:?}", err);
+    while let Some(blocks) = TokioStreamExt::next(&mut stream).await {
```

Comment:
has to explicitly define the `StreamExt` being used, as there are 2 of them used in this module